### PR TITLE
db_bench: introduce --benchmark_read_rate_limit

### PR DESCRIPTION
**Pull request by siying**
Summary: Add the parameter in db_bench to help users to measure latency histogram with constant read rate.

Test Plan: Run benchmark readrandom, seekranom, readwhilewriting, seekrandomwhilewriting with the setting, using one or multiple DBs.

### CODE REVIEW

File: tools/db_bench_tool.cc [hunk 1]
```diff
@@ -1821,6 +1827,8 @@ class Benchmark {
              / 1048576.0));
     fprintf(stdout, "Write rate: %" PRIu64 " bytes/second\n",
             FLAGS_benchmark_write_rate_limit);
+    fprintf(stdout, "Read rate: %" PRIu64 " bytes/second\n",
```
Comment (IslamAbdelRahman) [inline tools/db_bench_tool.cc]: from the code it looks to me that this should be ops/second instead of bytes/second

File: tools/db_bench_tool.cc [hunk 2]
```diff
@@ -3630,6 +3642,10 @@ class Benchmark {
       thread->stats.FinishedOps(nullptr, db, 1, kRead);
       ++i;
     }
+    if (thread->shared->read_rate_limiter.get() != nullptr) {
+      thread->shared->read_rate_limiter->Request(i, Env::IO_HIGH);
```
Comment (IslamAbdelRahman) [inline tools/db_bench_tool.cc]: Should not this be inside the loop ?

File: tools/db_bench_tool.cc [hunk 3]
```diff
@@ -3656,6 +3672,9 @@ class Benchmark {
...
```
Comment (IslamAbdelRahman) [inline tools/db_bench_tool.cc]: same

File: tools/db_bench_tool.cc (hunk 2 above)
Comment (siying) [inline tools/db_bench_tool.cc]: I'm requesting `i` here.

File: tools/db_bench_tool.cc (hunk 3 above)
Comment (siying) [inline tools/db_bench_tool.cc]: Same. `i` is requested. Not 1.

File: tools/db_bench_tool.cc (hunk 2 above)
Comment (IslamAbdelRahman) [inline tools/db_bench_tool.cc]: Yes, does that mean that the whole penalty will be at the end ? do we care about having smooth penalty ?

File: tools/db_bench_tool.cc (hunk 3 above)
Comment (siying) [inline tools/db_bench_tool.cc]: Maybe we should skip this in `readseq` test. It doesn't fit here very well.

Comment (siying) [inline tools/db_bench_tool.cc]: Now I get what you mean. OK. I'll move it inside the loop.

**Comment by siying** [comment]
Address comments.

File: tools/db_bench_tool.cc [hunk 4]
```diff
@@ -3655,6 +3673,13 @@ class Benchmark {
       bytes += iter->key().size() + iter->value().size();
       thread->stats.FinishedOps(nullptr, db, 1, kRead);
       ++i;
+      if (thread->shared->read_rate_limiter.get() != nullptr &&
+          i % 1024 == 1023) {
+        thread->shared->read_rate_limiter->Request(1024, Env::IO_HIGH);
+      }
+    }
+    if (thread->shared->read_rate_limiter.get() != nullptr) {
```
Comment (IslamAbdelRahman) [inline tools/db_bench_tool.cc]: looks like we forgot to remove this ?

Comment (siying) [inline tools/db_bench_tool.cc]: Good catch! Let me remove it.

**Comment by siying** [comment]
Tested --benchmarks=readreverse to verify the fix.

**Pull request closed without merging**